sql/core/src/main/scala/org/apache/spark/sql/columnar/ColumnType.scala

```diff
@@ -34,7 +34,8 @@ import org.apache.spark.unsafe.types.UTF8String
  *
  * @tparam JvmType Underlying Java type to represent the elements.
  */
-private[sql] sealed abstract class ColumnType[JvmType] {
+private[sql]
+sealed abstract class ColumnType[@specialized(Boolean, Byte, Short, Int, Long) JvmType] {
```

Comment:
Have we checked the compiled byte code to confirm this does help eliminating boxing?

@cloud-fan For `Float` and `Double`, had offline discussion with @davies. It's because this specialization is for simplifying `ColumnType.copyField`, which is only used in RLE encoder, and RLE encoder doesn't support `Float` or `Double` columns.